streams/src/main/java/org/apache/kafka/streams/state/internals/AbstractRocksDBSegmentedBytesStore.java

```diff
@@ -264,7 +264,6 @@ public void put(final Bytes key,
         final S segment = segments.getOrCreateSegmentIfLive(segmentId, context, observedStreamTime);
         if (segment == null) {
             expiredRecordSensor.record(1.0d, context.currentSystemTimeMs());
-            LOG.warn("Skipping record for expired segment.");
```

Comment:
is there a value in converting to debug instead of removing it?

Reply:
I don't see much value? We don't log any detailed information... -- the log line only gives you a processing time timestamp but nothing more...

If we would want to make it useful, we could add offset/timestamp metadata (but not log key/value as we never log data...) -- for this case, it might be useful to keep at DEBUG or TRACE.

Let's hear from others about it.

Reply:
I'm indifferent. We can convert it if anyone finds value in it but I don't really see it. But I also don't think it will crowd the debug logs too much either. 🤷